wagtaildraftail/client/registry.js

```diff
@@ -0,0 +1,41 @@
+const _registry = {
+  decorators: {},
+  sources: {},
+  strategies: {},
+};
+
+const registerDecorator = (name, decorator) => {
+  _registry.decorators[name] = decorator;
+  console.log('decorator: ' + name);
+};
```

Comment:
I think the signature of those `register` functions could be made simpler by taking advantage of object shorthands. New signature:

```js
const registerDecorators = (decorators) => {
  Object.assign(_registry.decorators, decorators);
};

window.wagtaildraftail.registerDecorators({ ButtonDecorator });
```

This has the advantage that you don't have to repeat the decorator name and its value. The other advantage is that it promotes registering all of your decorators / strategies / etc at once – we want to encourage people to separate the definition of those components from the side effect of registering them. This would look like:

```js
// In the JS entry bundle.
import ButtonDecorator from './ButtonDecorator';
import LocationDecorator from './LocationDecorator';

window.wagtaildraftail.registerDecorators({
   ButtonDecorator,
   LocationDecorator,
});
```